frontend/src/components/bucket/BucketPermission.vue

```diff
@@ -33,112 +33,115 @@ const removeBucketUser = (userId: string) => {
-      </template>
-    </Column>
-    <Column
-      header="Delete"
-      body-class="content-center"
-    >
-      <template #body="{ data }">
-        <Checkbox
-          v-model="data.delete"
-          input-id="delete"
-          :binary="true"
-          @input="(value) => updateBucketPermission(value, data.userId, Permissions.DELETE)"
-        />
-      </template>
-    </Column>
-    <Column
-      header="Manage"
-      body-class="content-center"
-    >
-      <template #body="{ data }">
-        <Checkbox
-          v-model="data.manage"
-          input-id="manage"
-          :binary="true"
-          @input="(value) => updateBucketPermission(value, data.userId, Permissions.MANAGE)"
-        />
-      </template>
-    </Column>
-    <Column header="Remove">
-      <template #body="{ data }">
-        <Button
-          class="p-button-lg p-button-text"
-          style="color: red"
-          @click="removeBucketUser(data.userId)"
-        >
-          <font-awesome-icon icon="fa-solid fa-user-xmark" />
-        </Button>
-      </template>
-    </Column>
-  </DataTable>
+      <Column
+        field="fullName"
+        header="Name"
+      />
+      <Column
+        header="Upload"
+        body-class="content-center"
+      >
+        <template #body="{ data }">
+          <Checkbox
+            v-model="data.create"
+            input-id="create"
+            :binary="true"
+            @input="(value) => updateBucketPermission(value, data.userId, Permissions.CREATE)"
+          />
+        </template>
+      </Column>
+      <Column
+        header="Read"
+        body-class="content-center"
+      >
+        <template #body="{ data }">
+          <Checkbox
+            v-model="data.read"
+            input-id="read"
+            :binary="true"
+            @input="(value) => updateBucketPermission(value, data.userId, Permissions.READ)"
+          />
+        </template>
+      </Column>
+      <Column
+        header="Update"
+        body-class="content-center"
+      >
+        <template #body="{ data }">
+          <Checkbox
+            v-model="data.update"
+            input-id="update"
+            :binary="true"
+            @input="(value) => updateBucketPermission(value, data.userId, Permissions.UPDATE)"
+          />
+        </template>
+      </Column>
+      <Column
+        header="Delete"
+        body-class="content-center"
+      >
+        <template #body="{ data }">
+          <Checkbox
+            v-model="data.delete"
+            input-id="delete"
+            :binary="true"
+            @input="(value) => updateBucketPermission(value, data.userId, Permissions.DELETE)"
+          />
+        </template>
+      </Column>
+      <Column
+        header="Manage"
+        body-class="content-center"
+      >
+        <template #body="{ data }">
+          <Checkbox
+            v-model="data.manage"
+            input-id="manage"
+            :binary="true"
+            :disabled="!data.elevatedRights"
```

Comment:
nice!

Reply:
i hope someone cant edit the html (eg remove `disabled="disabled"` or whatever it is) to enable this checkbox (?)

Reply:
Even if they did that, if the user in question attempting to edit permissions lacks the manage permissions permission, it should reject on COMS iirc.

Reply:
oh yeah. but for giving a user (who might not have data.elevatedRights) there's no check in COMS

Reply:
That's fine - elevatedRights is a BCBox concept, not a COMS concept. The permissions quintuplet should hold.

Reply:
i'm now thinking protecting against this kind of thing is out of scope for now.. and i'm nnot sure how yu would at this point.. and they can do worse by calling the COMS api directly..